scrapy/loader/__init__.py

```diff
@@ -26,7 +26,10 @@ class ItemLoader(object):
 
     def __init__(self, item=None, selector=None, response=None, parent=None, **context):
         if selector is None and response is not None:
-            selector = self.default_selector_class(response)
+            if response.selector.__class__ == Selector:
```

Comment:
Hi @nctl144,

There are two issues with this line:

1. If a subclass overrides `default_selector_class` attribute, it won't have an effect after this change. 
2. As I mentioned in https://github.com/scrapy/scrapy/issues/3125#issuecomment-366316774, this approach has an issue: when someone accesses `response.selector` first time, Scrapy parses response body; when this condition is False, we'd be parsing response body twice (here + later, at `self.default_selector_class(response)` line). This looks unnecesary.